lambdas/thumbnail/index.py

```diff
@@ -218,58 +232,75 @@ def lambda_handler(request):
     # Parse request info
     url = request.args['url']
     size = SIZE_PARAMETER_MAP[request.args['size']]
+    input_ = request.args.get('input', 'image')
     output = request.args.get('output', 'json')
+    page = request.args.get('page', 1)
 
     # Handle request
     resp = requests.get(url)
     if resp.ok:
-        # Get the original reader / format
-        # If the original reader isn't in the supported formats map, use PNG as default presentation format
         try:
             thumbnail_format = SUPPORTED_BROWSER_FORMATS.get(imageio.get_reader(resp.content), "PNG")
-        # In the case imageio can't read the image, default to PNG
-        # Usually an OME-TIFF / CZI / some other bio-format
         except ValueError:
-            thumbnail_format = "PNG"
-
-        # Read image data
-        img = AICSImage(resp.content)
-        orig_size = list(img.reader.data.shape)
-
-        # Generate a formatted ndarray using the image data
-        # Makes some assumptions for n-dim data
-        img = format_aicsimage_to_prepped(img)
-
-        # Send to Image object for thumbnail generation and saving to bytes
-        img = Image.fromarray(img)
-
-        # Generate thumbnail
-        img.thumbnail(size)
-        thumbnail_size = img.size
-
-        # Store the bytes
-        thumbnail_bytes = BytesIO()
-        img.save(thumbnail_bytes, thumbnail_format)
-
-        # Get bytes data
-        data = thumbnail_bytes.getvalue()
-
-        # Create metadata object
-        info = {
-            'original_size': orig_size,
-            'thumbnail_format': thumbnail_format,
-            'thumbnail_size': thumbnail_size,
-        }
+            thumbnail_format = "JPEG" if input_ == "pdf" else "PNG"
+        if input_ == "pdf":
```

Comment:
Just curious, does reading the PDF require Java to be installed on the image? A lot of document type file readers have Java deps in my experience.